zds/searchv2/views.py

```diff
@@ -328,17 +328,18 @@ def get_queryset_multisearch(self, search_requests, collection_names):
                 for entry in results[k]["hits"]:
                     entry["collection"] = collection_names[k]
                     all_collection_result.append(entry)
-        all_collection_result.sort(key=lambda result: result["text_match"], reverse=True)
-        count_results = len(all_collection_result)
-        if count_results > 10:
-            for i in range(int(count_results / 10)):
-                print(all_collection_result[i * 10 : (i + 1) * 10])
-                all_collection_result[i * 10 : (i + 1) * 10] = sorted(
-                    all_collection_result[i * 10 : (i + 1) * 10],
-                    key=lambda result: result["document"]["score"],
-                    reverse=True,
-                )
-                print(all_collection_result[i * 10 : (i + 1) * 10])
+        if self.search_query != "*":
```

Comment:
Je suis sceptique. J'allais dire que le bug se produit sans doute aussi avec d'autres caractères spéciaux, mais je n'ai pas réussi à en trouver. D'ailleurs, ça fonctionne si on peut un espace avant ou après l'astérisque. L'astérisque a une signification particulière pour TypeSense ? 

Une meilleure façon (à mon sens) de gérer ça serait plutôt de vérifier que `text_match` est bien présent. Qu'est-ce que ça veut dire s'il n'est pas présent ?

Reply:
Ok. Dans ce cas, il faudrait faire les choses suivantes : 
- ajouter un commentaire qui explique que `*` permet de retourner tous les documents avec TypeSense (et éventuellement mettre un lien vers la doc qui le dit)
- qu'est-ce qu'on fait si `*` est tout de même recherché ?  j'ai envie de dire de n'afficher aucun résultat (voire un message plus explicite comme `Recherche invalide.`)
- ajouter un test qui recherche `*`
- ce serait quand même bien de ne garder que les résultats qui ont un `text_match`, pour s'assurer que ce point ne nous pétera jamais à la figure. C'est faisable en conditionnant les deux lignes du `for` juste au-dessus par une condition `if 'text_match' in entry:`.

Reply:
Ah, je vois que vous vous servez dans les tests du fait que chercher `*` renvoie tout ce qui est indexé. Autant pour les tests, ça ne me gêne pas, autant en production la possibilité de demander à TypeSense de renvoyer tout ce qu'il a m'inquiète car c'est inutile et ça va utiliser beaucoup de ressource (bande passante, traitement des résultats, etc). À l'extrême on pourrait presque considérer comme une possible attaque par déni de service. Est-ce qu'il n'y a pas un autre moyen dans les tests de demander à TypeSense qu'il retourne tout ce qu'il a indexé ?